pkg/database/repository.go

```diff
@@ -57,31 +66,33 @@ func (r repository) Lock(id, instanceId, userId uint) (*model.Lock, error) {
 			Preload("Lock").
 			First(&d, id).Error
 		if err != nil {
+			if errors.Is(err, gorm.ErrRecordNotFound) {
+				err = errdef.NewNotFound("database not found by id: %d", id)
+			}
 			return err
 		}
 
 		if d.Lock != nil && d.Lock.InstanceID != 0 {
-			return fmt.Errorf("database already locked by user \"%d\" and instance \"%d\"", userId, d.Lock.InstanceID)
+			return errdef.NewBadRequest("database already locked by user \"%d\" and instance \"%d\"", userId, d.Lock.InstanceID)
 		}
 
 		lock = &model.Lock{
 			DatabaseID: id,
 			InstanceID: instanceId,
 			UserID:     userId,
 		}
-		err = tx.Create(lock).Error
-		if err != nil {
-			return err
-		}
-
-		return nil
+		return tx.Create(lock).Error
 	})
 
 	return lock, errTx
 }
 
 func (r repository) Unlock(id uint) error {
-	return r.db.Delete(&model.Lock{}, id).Error
+	err := r.db.Delete(&model.Lock{}, id).Error
+	if errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
Gorm delete does not seem to return ErrRecordNotFound if what we are trying to delete does not exist https://stackoverflow.com/questions/67154864/how-to-handle-gorm-error-at-delete-function

Reply:
Good catch!